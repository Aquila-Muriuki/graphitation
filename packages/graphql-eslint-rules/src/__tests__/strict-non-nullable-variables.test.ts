import {
  GraphQLRuleTester,
  ParserOptions,
} from "@graphql-eslint/eslint-plugin";
import rule from "../strict-non-nullable-variables";

const ruleTester = new GraphQLRuleTester();

const TEST_SCHEMA = /* GraphQL */ `
  type Query {
    sum(numbers: [Int!]!): Int
    sumSafe(numbers: [Int]!): Int
    addOne(number: Int!): Int
    addOneSafe(number: Int): Int
    addTwo(input: AddTwoInput!): Int
    addTwoSafe(input: AddTwoSafeInput!): Int
  }

  input AddTwoInput {
    left: Int!
    right: Int!
  }

  input AddTwoSafeInput {
    left: Int
    right: Int
  }
`;

const parserOptions: ParserOptions = {
  schema: TEST_SCHEMA,
  filePath: "test.graphql",
};

ruleTester.runGraphQLTests("strict-non-nullable-variables", rule, {
  valid: [
    {
      code: `
query Q ($number: Int! = 3) {
  sum(numbers: [1, $number, 3])
}`,
      parserOptions,
    },
    {
      code: `
query Q ($number: Int = 3) {
  sumSafe(numbers: [1, $number, 3])
}`,
      parserOptions,
    },
    {
      code: `
query Q ($number: Int! = 3) {
  addOne(number: $number)
}`,
      parserOptions,
    },
    {
      code: `
query Q ($number: Int = 3) {
  addOneSafe(number: $number)
}`,
      parserOptions,
    },
    {
      code: `
query Q ($number: Int! = 3) {
  addTwo(input: { left: $number, right: 4 })
}`,
      parserOptions,
    },
    {
      code: `
query Q ($number: Int = 3) {
  addTwoSafe(input: { left: 2, right: $number })
}`,
      parserOptions,
    },
  ],
  invalid: [
    {
      code: `
query Q ($number: Int = 3) {
  sum(numbers: [1, $number, 3])
}`,
      parserOptions,
      errors: 1,
    },
    {
      code: `
query Q ($number: Int = 3) {
  addOne(number: $number)
  addOneSafe(number: $number)
}`,
      parserOptions,
      errors: 1,
    },
    {
      code: `
query Q ($number: Int = 3) {
  addTwo(input: { left: $number, right: 2 })
}`,
      parserOptions,
      errors: 1,
    },
  ],
});
