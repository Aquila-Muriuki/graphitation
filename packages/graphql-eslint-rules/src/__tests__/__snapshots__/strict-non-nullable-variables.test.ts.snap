// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | query Q ($number: Int = 3) {
      2 |   sum(numbers: [1, $number, 3])
      3 | }

#### ❌ Error

      1 | query Q ($number: Int = 3) {
    > 2 |   sum(numbers: [1, $number, 3])
        |                    ^ Can't use nullable variable in non nullable position.
      3 | }"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | query Q ($number: Int = 3) {
      2 |   addOne(number: $number)
      3 |   addOneSafe(number: $number)
      4 | }

#### ❌ Error

      1 | query Q ($number: Int = 3) {
    > 2 |   addOne(number: $number)
        |                  ^ Can't use nullable variable in non nullable position.
      3 |   addOneSafe(number: $number)"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 | query Q ($number: Int = 3) {
      2 |   addTwo(input: { left: $number, right: 2 })
      3 | }

#### ❌ Error

      1 | query Q ($number: Int = 3) {
    > 2 |   addTwo(input: { left: $number, right: 2 })
        |                         ^ Can't use nullable variable in non nullable position.
      3 | }"
`;
